src/main/scala/ai/privado/entrypoint/DynamicRuleMerger.scala

```diff
@@ -0,0 +1,44 @@
+package ai.privado.entrypoint
+
+import ai.privado.model.{ConfigAndRules, RuleInfo}
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable.ListBuffer
+
+object DynamicRuleMerger {
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  def mergeDynamicRuleSinkForDependencyDiscovery(
```

Comment:
Any reason to create this as separate Object `DynamicRuleMerger` instead of adding this method `mergeDynamicRuleSinkForDependencyDiscovery` inside `RuleProcessor`?

Reply:
In my opinion, it should be part of `RuleProcessor` class.

Reply:
Yes we can. My thought was to separate the merging utility from the `ruleProcessor` because, in the future, there will be more merging logic for `storage`, `API`, and their special cases.

Reply:
I propose to create a `trait DynamicRuleMerger` and we can use the functions of this class as `trait RuleProcessor extends DynamicRuleMerger`

Reply:
Yes right using `trait` would fulfil out requirement.

Reply:
done.